lib/modules/console_listener/index.js

```diff
@@ -0,0 +1,79 @@
+const utils = require('../../utils/utils.js');
+
+class ConsoleListener {
+    constructor(embark, options) {
+        this.logger = embark.logger;
+        this.ipc = options.ipc;  
+        this.addressToContract = [];
+        this.contractsConfig = embark.config.contractsConfig;
+        this.listenForLogRequests();
+    }
+
+    _updateContractList(){
+        Object.keys(this.contractsConfig.contracts).forEach(contractName => {
+            let contract = this.contractsConfig.contracts[contractName];
+            let address = contract.deployedAddress.toLowerCase();
+
+            if(!this.addressToContract[address]){
+                let funcSignatures = {};
+                contract.abiDefinition
+                    .filter(func => func.type == "function")
+                    .map(func => {
+                                return {
+                                    name: func.name + 
+                                            '(' + 
+                                            (func.inputs ? func.inputs.map(input => input.type).join(',') : '') +
+                                            ')',
+                                    abi: func,
+                                    functionName: func.name
+                                };
+                        })
+                    .forEach(func => {
+                        funcSignatures[utils.sha3(func.name).substring(0, 10)] = func;
+                    });
+
+                this.addressToContract[address] = {
+                    name: contractName,
+                    functions: funcSignatures
+                };
+            }
+        });
+    }
+
+    listenForLogRequests(){
```

Comment:
This function should have the `_` before (`_listenForLogRequests`) too it is clearly private (only itself should setup the listening (we don't want multiple listens))